### ISSUE TITLE
[REQUEST] More realistic starting positions for AIRCOND knobs 

### ISSUE THREAD

**Issue by aviationenthusiast** [flybywiresim/a32nx#3128]
**Is your feature request related to a problem? Please describe.**

All three of the air conditioning knobs are always in the same starting position C&D (now 12 o'clock). That doesn't account for the small variations that happen from situation to another, and from pilot to pilot.

**Describe the feature you'd like to see implemented**

1- Instead of a fixed starting position for all three knobs, we could implement a **range** of positions, one of which gets selected when loading the plane. Either randomly, or based on distance from the equator/seasons or both, according to what IRL pilots find works best.

2- The 3 knobs shouldn't have the same exact starting value either, as if they were last turned by a robotic hand. So maybe a smaller range among the 3 knobs in the **same flight** as well.

**References**

According to @ImenesFBW in PR #3118 "Pilots have been giving feedback about how they rarely touch the air conditioning knobs and **usually** find them in the **10-12 o'clock** position in a cold&dark cockpit."

**Additional context**

3- If pilots reported that there is a 10% chance the knobs could be **outside** the 10-12 range for example, it would be nice to account for that as well.

**Comment by Benjozork** [comment]
This is not easily possible using a simple aircraft state file.

**Comment by beheh** [comment]
Fixed by #3118, no?

**Comment by ErikNL80** [comment]
Why not putting in a fixed variation between 10 and 12 o'clock? This way it looks more random... Personally I think dev time can be spent better than programming a random position for an airco knob.

**Comment by aviationenthusiast** [comment]
> 
> 
> Why not putting in a fixed variation between 10 and 12 o'clock? This way it looks more random... Personally I think dev time can be spent better than programming a random position for an airco knob.

A fixed position every time you load up the aircraft isn't the most realistic.
I respect your opinion. But the point of the mod is to make the experience as realistic as possible. So if something is still doable, then why not point it out until someone feels like doing it you know.

**Comment by beheh** [comment]
> Personally I think dev time can be spent better than programming a random position for an airco knob.

There are lots of developers on this project working on a lot of different things. There's also people who have never contributed, but wouldn't mind to. A small detail like this is great to get someone newly involved in the project.

**Comment by Rameyuk** [comment]
I like the current position at 12 o'clock now. That's one less thing for me to turn the knob when I start the flight. Previously I had to keep turning it everytime to the 12 o'clock position. And since the mousewheel turn isn't exactly precise, I have to use 2x additional clicks to get the knobs in the right position.

At least with the current implementation of always at 12 gives it the most optimal aircon temperature. And that's one less thing for me to do for optimising my startup procedure to be as concise as possible.

**Comment by davidwalschots** [comment]
I've tagged this as "Good First Issue". One can probably write some JavaScript code to randomly set these knobs at flight load. Most of the time they should be close to the 12 o'clock position, but all aircraft are different. If the 12 o'clock position would always work, the knobs wouldn't be there.

**Comment by rare-potato** [comment]
I'll claim this. I've already started working on it and I'm planning on making a draft PR for it within the next 24 hours if I'm able to.

**Comment by cptnuss-ops** [comment]
Is this issue still open or did the referenced PR fixed the issue?

**Comment by rare-potato** [comment]
> Is this issue still open or did the referenced PR fixed the issue?

Issue is still open, referenced PR was never finished and I had other things come up so never got around to coming back to it.

**Comment by cptnuss-ops** [comment]
> > Is this issue still open or did the referenced PR fixed the issue?
> 
> Issue is still open, referenced PR was never finished and I had other things come up so never got around to coming back to it.

Ok, then I can work on this issue?

**Comment by rare-potato** [comment]
> Ok, then I can work on this issue?

Yep. I'd suggest you at least look at the changes I had worked on with PR #6922. They may be of use and could make it easier/faster for you to complete, or it could be completely useless 🤣

**Comment by cptnuss-ops** [comment]
> > Ok, then I can work on this issue?
> 
> 
> 
> Yep. I'd suggest you at least look at the changes I had worked on with PR #6922. They may be of use and could make it easier/faster for you to complete, or it could be completely useless 🤣 

Sounds good, I'll have a look 😃

**Comment by noelpferd** [comment]
Hello,
I implemented a script for realizing some random panel states at startup. Maybe it's something for you: [https://github.com/noelpferd/MSFS-FBW-Panel-Randomizer](https://github.com/noelpferd/MSFS-FBW-Panel-Randomizer)
